x-pack/plugins/fleet/server/services/epm/kibana/index_pattern/install.ts

```diff
@@ -85,34 +89,39 @@ export async function installIndexPatterns(
   );
 
   const packagesToFetch = installedPackagesSavedObjects.reduce<
-    Array<{ name: string; version: string; installSource: InstallSource }>
+    Array<{ name: string; version: string; installedPkgSO: SavedObject<Installation> | undefined }>
   >((acc, pkgSO) => {
     acc.push({
       name: pkgSO.attributes.name,
       version: pkgSO.attributes.version,
-      installSource: pkgSO.attributes.install_source,
+      installedPkgSO: pkgSO,
     });
     return acc;
   }, []);
 
   if (pkgName && pkgVersion && installSource) {
-    const packageToInstall = packagesToFetch.find((pkgSO) => pkgSO.name === pkgName);
+    const packageToInstall = packagesToFetch.find((pkg) => pkg.name === pkgName);
     if (packageToInstall) {
       // set the version to the one we want to install
       // if we're reinstalling the number will be the same
       // if this is an upgrade then we'll be modifying the version number to the upgrade version
       packageToInstall.version = pkgVersion;
     } else {
       // if we're installing for the first time, add to the list
-      packagesToFetch.push({ name: pkgName, version: pkgVersion, installSource });
+      packagesToFetch.push({
+        name: pkgName,
+        version: pkgVersion,
+        installedPkgSO: await getInstallationObject({ savedObjectsClient, pkgName }),
```

Comment:
Do we need to pass the Saved Object or will a simple `Installation` do? AFAICT, we only use the `.attributes` properties of the SO, so I think `getInstallationObject` would work and save all the `.attributes` access later

Reply:
Cool! Wherever possible I'd like to use `installedPkg: Installation` instead of  `installedPkgSO: SavedObject<Installation>`